torch/distributed/rpc/internal.py

```diff
@@ -72,6 +72,7 @@ def serialize(self, obj):
         # in python. also, when _internal_rpc_pickler is imported to rpc/api.py, rpc.RRef is not
         # compiled yet, it is not good place to acces rpc.RRef inside _InternalRPCPickler constructor,
         # so puting rref's dispatch table here
+        p.dispatch_table[dist.rpc.PyRRef] = self._rref_reducer
```

Comment:
can you elaborate why we need this PyRRef dispatch here? I assume every object's type from the python land is just using `rpc.RRef`?

Reply:
The deserialized RRef object on an RPC receiver side will be in type of `rpc.PyRRef`. (returned from C++ pybind)

The return value of a `rpc.remote(..)` call is of `rpc.PyRRef` type. (returned from C++ pybind)

Only the RRef created locally by RRef constructor is of `rpc.RRef` type.